src/test/java/com/mekomsolutions/eip/route/OdooPatientHandlerSyncsAnyCreatedPatientEventRouteTest.java

```diff
@@ -0,0 +1,130 @@
+
+import ch.qos.logback.classic.Level;
+
+@TestPropertySource(properties = {"openmrs.identifier.type.uuid=" + OdooPatientHandlerSyncsAnyCreatedPatientEventRouteTest.ID_TYPE_UUID, "should.sync.any.patient.asCustomer=true"})
+public class OdooPatientHandlerSyncsAnyCreatedPatientEventRouteTest extends BaseOdooRouteTest {
+	
+	private static final String ROUTE_ID = "odoo-patient-handler";
+	
+	public static final String EX_PROP_PATIENT = "patient";
+	
+	public static final String EX_PROP_CREATE_CUSTOMER = "createCustomerIfNotExist";
+	
+	public static final String EX_PROP_CUSTOM_DATA = "customPatientData";
+	
+	public static final String ID_TYPE_UUID = "8d79403a-c2cc-11de-8d13-0010c6dffd0f";
+	
+	public static final String ID_TYPE_ID_KEY = "odoo-patient-handler-idTypeId";
+	
+	@EndpointInject("mock:odoo-get-customer")
+	private MockEndpoint mockGetCustomerEndpoint;
+	
+	@EndpointInject("mock:odoo-manage-customer")
+	private MockEndpoint mockManageCustomerEndpoint;
+	
+	@EndpointInject("mock:odoo-get-quotations")
+	private MockEndpoint mockGetQuotesEndpoint;
+	
+	@EndpointInject("mock:odoo-cancel-quotations")
+	private MockEndpoint mockCancelQuotesEndpoint;
+	
+	@EndpointInject("mock:odoo-process-person-address")
+	private MockEndpoint mockProcessAddressEndpoint;
+	
+	@EndpointInject("mock:odoo-callback-get-custom-customer-data")
+	private MockEndpoint mockGetCustomDataEndpoint;
+	
+	@Before
+	public void setup() throws Exception {
+		AppContext.remove(ID_TYPE_ID_KEY);
+		mockGetCustomerEndpoint.reset();
+		mockManageCustomerEndpoint.reset();
+		mockProcessAddressEndpoint.reset();
+		mockGetQuotesEndpoint.reset();
+		mockCancelQuotesEndpoint.reset();
+		mockGetCustomDataEndpoint.reset();
+		mockGetCustomerEndpoint.expectedMessageCount(1);
+		advise(ROUTE_ID, new AdviceWithRouteBuilder() {
+			
+			@Override
+			public void configure() {
+				interceptSendToEndpoint("direct:odoo-get-customer").skipSendToOriginalEndpoint().to(mockGetCustomerEndpoint);
+				interceptSendToEndpoint("direct:odoo-callback-get-custom-customer-data").skipSendToOriginalEndpoint()
+				        .to(mockGetCustomDataEndpoint);
+				interceptSendToEndpoint("direct:odoo-manage-customer").skipSendToOriginalEndpoint()
+				        .to(mockManageCustomerEndpoint);
+				interceptSendToEndpoint("direct:odoo-process-person-address").skipSendToOriginalEndpoint()
+				        .to(mockProcessAddressEndpoint);
+				interceptSendToEndpoint("direct:odoo-get-quotations").skipSendToOriginalEndpoint().to(mockGetQuotesEndpoint);
+				interceptSendToEndpoint("direct:odoo-cancel-quotations").skipSendToOriginalEndpoint()
+				        .to(mockCancelQuotesEndpoint);
+			}
+			
+		});
+	}
+	
+	@After
+	public void tearDown() throws Exception {
+		mockGetCustomerEndpoint.assertIsSatisfied();
+	}
+	
+	@Test
+	public void shouldAddPatientToOdooIfTheyDoNotExistAddCreateCustomerPropIsSetToFalseGivenshouldSyncAnyPatientAsCustomerIsTrue() throws Exception {
```

Comment:
You also need a control test that ensures that if the property is not set to true the patient doesn't get pushed to odoo

Reply:
Ok...

Reply:
However wouldn't the existing tests in `OdooPatientHandlerTest` act as control tests?

Reply:
They can as long as they exist